runtime/src/bank.rs

```diff
@@ -4545,6 +4597,9 @@ impl Bank {
             },
         );
 
+        if let Some(pre_commit_callback) = pre_commit_callback {
+            pre_commit_callback(timings, &processing_results)?;
+        }
         let commit_results = self.commit_transactions(
```

Comment:
@ryoqun. I think we may need to grab the `freeze_lock` here, similar to how we do in block-production.

`commit_transactions` requires that the bank is not frozen yet (obviously), but there's a small gap between recording and this line where the bank could get frozen.

Reply:
ugh, quite good find... fixed: fb71c0dccaab66db36228f0f3ab45e12c0d4377d